### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by tafodinho**

### CODE REVIEW

File: methods_to_test.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+const capitalize = (string) => {
+    return string.charAt(0).toUpperCase() + string.slice(1);
+}
+
+const reverseString = (string) => {
+    let splitString = string.split("")
```
Comment (rvvergara) [inline methods_to_test.js:6]: [OPTIONAL]
We should prefer the use of `const` over `let`. Had we used a linter this would have been given to us as a warning.

File: methods_to_test.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    let reversedArray = splitString.reverse()
+    let joinString = reversedArray.join("")
+
+    return joinString
+}
+
+const analyse = (numbers) => {
+    let sum = numbers.reduce((previous, current) => current += previous) / numbers.length
+    // let avrg = sum / numbers.length
```
Comment (rvvergara) [inline methods_to_test.js:15]: Remove commented code please

File: methods_to_test.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+const reverseString = (string) => {
+    let splitString = string.split("")
+    let reversedArray = splitString.reverse()
+    let joinString = reversedArray.join("")
+
+    return joinString
+}
+
+const analyse = (numbers) => {
+    let sum = numbers.reduce((previous, current) => current += previous) / numbers.length
```
Comment (rvvergara) [inline methods_to_test.js:14]: Why do we call this "sum" ? We are computing for the average. Let us use a more meaningful variable name.

Comment (rvvergara) [inline methods_to_test.js:14]: Why do we need to `current += previous`. Can't we simply `current + previous` ?

Comment (rvvergara) [inline methods_to_test.js:14]: use `const`

File: methods_to_test.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        min: Math.min(...numbers),
+        max: Math.max(...numbers),
+        length: numbers.length
+    }
+}
+
+class Calculator {
+
+    add(...numbers) {
+        return numbers.reduce((prev, cur) => cur *= prev)
```
Comment (rvvergara) [inline methods_to_test.js:28]: I am extremely sure that this operation isn't "adding" numbers.

I think the test case we wrote for this has played a trick on us. :D

File: methods_to_test.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    divide(a, b) {
+        return a / b
+    }
+    multiply(...numbers) {
+        return numbers.reduce((prev, cur) => cur *= prev)
+    }
+
+}
+
+class Cypher {
```
Comment (rvvergara) [inline methods_to_test.js:44]: why do we need the `caesarCypher` to be a `class` ? Let us convert this to a regular function only.

File: methods_to_test.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    shiftChar(char) {
+        if(char === " ") {
+            return char
+        }
+        return Object.keys(this.defaultString)[(this.defaultString[char] + 5)%26] 
+    }
+}
+
+const calc = new Cypher(5)
+console.log(calc.encypher("Hell56o loo!k who we got here"))
```
Comment (rvvergara) [inline methods_to_test.js:67]: Remove `console.log` code.

File: methods_test.test.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    expect(capitalize("name")).toMatch(/[N]ame/)
+})
+
+test("reverse string", () => {
+    expect(reverseString("string")).toMatch(/gnirts/)
+})
+
+test("calculator add", () => {
+    const calc = new Calculator
+    expect(calc.add(1,2,3)).toBe(6)
```
Comment (rvvergara) [inline methods_test.test.js:15]: This test is deceptive, why? It deceives us into believing our `add` function gives the correct answer. In fact, it gives you the wrong answer if we are to consider it should add the numbers.

```javascript
1 + 2 + 3 === 1 * 2 * 3 // will give you true
```
In short regardless of whether we use `add` or `multiply` to `1, 2, 3` as the three arguments it will always return 6. 

This is an evidence that we should write **MORE THAN ONE** test case for each function/unit we are testing. Let us write at least 3 cases.

File: methods_test.test.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
+// import capitalize from './methods_to_test';
+const { capitalize, reverseString, analyse, Calculator, Cypher } = require('./methods_to_test.js')
+
+
+test("capitalize", () => {
```
Comment (rvvergara) [inline methods_test.test.js:5]: ## CRUCIAL

Let us write at least 3 test cases (considering edge cases) for each function. We cannot rely that our functions/methods are correct just because it passes one test.

**Review by rvvergara — CHANGES_REQUESTED** [review]
## Hi @tafodinho 

## Here are some instructions that I think we didn't follow
1. This project should be collaborative. We are doing it solo. Kindly explain why.
2. We are to use [Git Flow](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/) here. We are to make pull requests from a feature branch into `development`. So what we should have done here was create a branch called (something like) `feature/testing`. 
3. We should use a [linter](https://github.com/microverseinc/linters-config/tree/master/javascript) here.
  - `stickler` should have been activated for this project
  - `eslint` should have been installed

### Let us not rush through the projects
I have been reviewing your react calculator project and I don't think there has been a thumbs up so far. Then I also see that you submitted a review request for the redux bookstore. Those two react projects should have not been started at all actually since you still need to pass this Testing Practice project as well as the succeeding project. Kindly discuss with Student Success regarding this. I don't think we are doing things the right way. The danger is that the concepts we need to learn are not sinking in properly. It's like basketball, let us not attempt to do a tomahawk dunk when we haven't mastered shooting some layups.

**Comment by tafodinho** [comment]
@rvvergara my partner was unavailable for that time

File: methods_to_test.js (hunk 5 above)
Comment (tafodinho) [reply methods_to_test.js:44]: from the instructions it suits the Cypher to be a class more
